includes/Modules/AdSense.php

```diff
@@ -247,6 +267,24 @@ function() use ( $client_id ) {
 		$this->adsense_tag_printed = true;
 	}
 
+	/**
+	 * Outputs the <amp-auto-ads> tag.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $client_id AdSense client ID to use in the snippet.
+	 */
+	protected function print_amp_auto_ads( $client_id ) {
+		if ( $this->adsense_tag_printed ) {
+			return;
+		}
+
+		?>
+		<amp-auto-ads type="adsense" data-ad-client="<?php echo esc_attr( $client_id ); ?>"></amp-auto-ads>
+		<?php
+		$this->adsense_tag_printed = true;
```

Comment:
Minor detail, but I think it would be a bit cleaner to put this assignment right after the guard that checks it. We do it this way in several other places and it generally makes it a bit easier to read by separating the logic from output. The only reason I think we would want to set it afterwards was if there was the potential of an exception to be thrown during the render (potentially inside an output buffer), but that's not really the case here 🙂